bsdlib/doc/pdn_extension.rst

```diff
@@ -0,0 +1,174 @@
+.. _pdn_extension:
+
+PDN management
+##############
+
+The BSD library supports configuring up to 10 packet data network (PDN) sockets.
+PDN sockets are an extension to the standard socket API.
+They make it easy to configure PDN connections as identified by the user's Access Point Name (APN).
+
+After a PDN socket is configured, it can be used by a data socket (IP or RAW socket).
+The PDN socket dictates where to send the data, or to do DNS queries on the PDN that is associated with the given APN.
+
+The following sections give pseudo code for typical use cases of PDN sockets.
+
+Creating a PDN socket
+*********************
+
+.. code-block:: c
+
+   pdn_fd = nrf_socket(NRF_AF_LTE, NRF_SOCK_MGMT, NRF_NPROTO_PDN);
+   nrf_connect(pdn_fd, (struct nrf_sockaddr *)"custom.apn", strlen("custom.apn"));
+
+When creating a PDN socket, the library requests a context ID (CID) to be created.
+It then configures the CID, which creates a PDN.
+This PDN is used to handle data that is sent over the connection.
+
+The example code shows how to create a PDN socket with the protocol family NRF_NPROTO_PDN and the proprietary address family NRF_AF_LTE.
```

Comment:
This explains the code so it would be better if it immediately followed the snippet (see previous comment)